src/hooks/use-site-details.tsx

```diff
@@ -218,6 +218,7 @@ export function SiteDetailsProvider( { children }: SiteDetailsProviderProps ) {
 						id: tempSiteId,
 						name: siteName || path,
 						path,
+						port: 9999, // Set a temporary port
```

Comment:
Why we need to set temporary port?

Reply:
It's used as a placeholder to comply with the types.
This site placeholder is added so that Studio can display the new site in the sidebar while it's being created. The site information will be replaced once the site is created when the call to  `const data = await getIpcApi().createSite(path, siteName);` resolves.

We could split `getIpcApi().createSite` into two functions: one that retrieves the siteId and new port, add the site to the state in the react, and then pass those variables as parameters to the createSite function.

If we decide to use the temporary port, we could use a more different port like `-1` that could help us discover any issues during site creation.

Reply:
@sejas Updated to `-1`, makes sense, good idea 👍